libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Solo.cpp

```diff
@@ -43,10 +43,16 @@ void AP_BattMonitor_SMBus_Solo::timer()
     // read cell voltages
     if (read_block(BATTMONITOR_SMBUS_SOLO_CELL_VOLTAGE, buff, 8, false)) {
         float pack_voltage_mv = 0.0f;
+        float cell_diff_mv = 0.0f;
```

Comment:
Don't do this here, this logic is identical for all the SMBus batteries. This should either be done as a single loop in the general SMBus backend, or just computed on demand in the failsafe check (this would be my preference as we could remove the state then).

Reply:
I would rather it out in the failsafe check too, but I couldn't see where I'd get a cell count to use in the for loop.  What do you suggest for that?

Reply:
Reverted. Will think of something else.

Reply:
Done. Moved out to the state.